### PR TITLE
custom read/write test

### PR DESCRIPTION
**Pull request by stephenberry**

### CODE REVIEW

File: tests/json_test/json_test.cpp [hunk 1]
```diff
@@ -7086,6 +7086,48 @@ suite expected_tests = [] {
...
+         value.str += "write";
+         write<json>::op<Opts>(value.str, args...);
+      }
+   };
+}
+
+suite custom_struct_tests = [] {
+   "custom_struct"_test = [] {
+      custom_struct obj{};
+      std::string s{};
```
Comment (arturbac) [inline tests/json_test/json_test.cpp:7122]: try changing it to const

`std::string const s{};`

Comment (stephenberry) [reply tests/json_test/json_test.cpp:7122]: You can't use const for your buffer if you are writing, because it may need to grow the buffer.

Comment (arturbac) [reply tests/json_test/json_test.cpp:7122]: my bad, I was thinking about custom_struct obj{};
But I am busy at work and I made mistake

Comment (stephenberry) [reply tests/json_test/json_test.cpp:7122]: No problem

**Pull request merged**